### PR TITLE
components/Toast: When display an error, strongly suggest reporting it

### LINKED ISSUE #1024: Error notifications should suggest creating a debug bundle
(from https://github.com/keyboardio/Chrysalis/issues/989#issuecomment-1199643605)

> also, once i completed the update process, i got an error dialog telling me an error had occurred while flashing; otoh when i tried to repeat the process, the UI showed that i had the latest firmware version installed. it's not clear to me whether that error was safe to ignore, or what i would do to address it (factory reset and reflash?), or if i could retrieve the error message by creating a debug bundle.

While we have great logs nowadays, Chrysalis does not do a good job of reminding our users of the debug bundle. While we can ask them later, digging the logs out once Chrysalis has been closed is - while possible - considerably harder than just creating the bundle while the error event is still fresh.

Our error notifications should either mention the bundle, or better yet, provide a "Create debug bundle" button.

### PR DESCRIPTION
**Pull request by algernon**
When Toast displays an error notification, display a "Report the problem" button alongside it, which navigates to the "Report a problem" screen upon click.

Fixes #1024.

Screenshot from 2022-07-31 14-17-43

### CODE REVIEW

**Comment by obra** [comment]
if we have users reporting every communications time out, I suspect we’re gonna get a lot of spurious tickets. maybe, we should be linking to a list of open issues with instructions to open a new one if they don’t see theirs there. we should also only show the report an issue but if they are running the current release.

**Comment by obra** [comment]
alternatively, we could use something like sentry.io to log those errors automatically

**Comment by algernon** [comment]
> if we have users reporting every communications time out, I suspect we’re gonna get a lot of spurious tickets.

Maybe. But we also hear about a number of cases where there were genuine bugs, but no bundle made at the time. Perhaps we could compromise, and make it easier to access past logs, so we can ask our users to upload an existing log (without having to explain where they are on each platform), rather than ask them to repro and make a bundle?

> maybe, we should be linking to a list of open issues with instructions to open a new one if they don’t see theirs there.

The "report an issue" screen would be a good place for those instructions, in my opinion. The notification isn't a good place neither for instructions, nor for a direct link, imo.

> we should also only show the report an issue but if they are running the current release.

Can't reliably do, because Chrysalis should 100% work fine offline, and we can't verify we're the latest without pinging GitHub.

> alternatively, we could use something like sentry.io to log those errors automatically

While that would provide us a lot of information to work with, I'm not a fan of automatically uploading error logs to third party services, even if they're opt-in. At the moment, we'd need to upload *all* logs, because the error logs themselves are usually not enough to figure out what went wrong, there's not enough context. Making sure that the logs include enough context would be a tough job, and I don't think it would yield enough benefit to worth the effort.

We'd still have to wade through a lot of spurious sessions, so all we'd gain is having less tickets - but that's just the surface. We'd still have the same amount of data, with the same % of spurious sessions to go through.

TLDR: You convinced me that the approach in this PR is not the way to go. My next proposal would be to make it easier to access past logs, so when an issue is reported, we can ask for past logs too, if need be.

**Comment by algernon** [comment]
Reopening, we're going to do something similar, but put it behind a feature flag: "Show issue reporting button on all error messages" under developer tools.

**Comment by obra** [comment]
Now that Chrysalis is on the web, we're going to be able to log errors programatically.
